cognite/neat/_data_model/_analysis.py

```diff
@@ -1,154 +1,455 @@
+
+
+ResourceSource = Literal["auto", "local", "cdf", "both"]
+
+
+@dataclass
+class CDFSnapshot:
+    data_models: dict[DataModelReference, DataModelRequest] = field(default_factory=dict)
+    views: dict[ViewReference, ViewRequest] = field(default_factory=dict)
+    containers: dict[ContainerReference, ContainerRequest] = field(default_factory=dict)
+    spaces: dict[SpaceReference, SpaceRequest] = field(default_factory=dict)
+
+    @classmethod
+    def from_cdf(cls, client: NeatClient) -> "CDFSnapshot":
+        """Create CDFSnapshot from CDF API responses."""
+        return CDFSnapshot(
+            # TODO: spaces and data_models should be update after updating list methods for unlimited no
+            spaces={
+                response.as_reference(): response.as_request()
+                for response in client.spaces.list(include_global=True, limit=1000)
+            },
+            data_models={
+                response.as_reference(): response.as_request()
+                for response in client.data_models.list(all_versions=True, include_global=True, limit=1000)
+            },
+            views={
+                response.as_reference(): response.as_request()
+                for response in client.views.list(
+                    all_versions=True, include_global=True, include_inherited_properties=False, limit=None
+                )
+            },
+            containers={
+                response.as_reference(): response.as_request()
+                for response in client.containers.list(include_global=True, limit=None)
+            },
+        )
+
+
+@dataclass
+class LocalSnapshot:
+    """Local data model resources."""
+
+    data_model: DataModelRequest
+    views: dict[ViewReference, ViewRequest]
+    containers: dict[ContainerReference, ContainerRequest]
+
+    @classmethod
+    def from_request_schema(cls, schema: RequestSchema) -> "LocalSnapshot":
+        """Create LocalSnapshot from a DataModelRequest."""
+        return LocalSnapshot(
+            data_model=schema.data_model,
+            views={view.as_reference(): view for view in schema.views},
+            containers={container.as_reference(): container for container in schema.containers},
+        )
+
+
+class ValidationResources:
     def __init__(
-        self,
-        physical: RequestSchema | None = None,
+        self, modus_operandi: ModusOperandi, local: LocalSnapshot, cdf: CDFSnapshot, limits: SchemaLimits
     ) -> None:
-        self._physical = physical
-
-    @property
-    def physical(self) -> RequestSchema:
-        if self._physical is None:
-            raise ValueError("Physical Data Model is required for this analysis")
-        return self._physical
+        self.local = local
+        self.cdf = cdf
+        self.limits = limits
+        self._modus_operandi = modus_operandi
+
+        # Update local resources based on modus operandi
+        self._update_local_resources()
```

Comment:
This  mutates the local snapshot  if the modus operandi is addetive. 
It is also a bit magical, what about something like this, and then in the `LocalSnapshot.merge(...)` use model_copy such that you do not mutate the objects?

```suggestion
        if self._modus_operandi == "rebuild":
            self._merged = local
        elif self._modus_operandi == "additive":
            self._merged = local.merge(cdf)
        else:
            raise RuntimeError(f"ValidationResources: Unknown modus: {self._modus_operandi}. This is a bug!")
```

Reply:
Fair point !